cfme/tests/infrastructure/test_scvmm_specific.py

```diff
@@ -14,18 +14,23 @@
 
 
 @pytest.fixture
-def testing_vm_without_dvd(provider, small_template):
-    vm_name = "test_no_dvd_{}".format(fauxfactory.gen_alpha())
+def vm(provider, small_template):
+    vm_name = "test-scvmm-{}".format(fauxfactory.gen_alpha())
     vm = provider.appliance.collections.infra_vms.instantiate(
-        vm_name, provider, small_template.name)
-    vm.create_on_provider()
-    vm.mgmt.disconnect_dvd_drives()
+        vm_name, provider, small_template.name
+    )
+    vm.create_on_provider(find_in_cfme=True)
     yield vm
     vm.cleanup_on_provider()
 
 
+@pytest.fixture
+def remove_dvd_from_vm(vm):
+    vm.mgmt.disconnect_dvd_drives()
```

Comment:
Q:
It will work as a single use... but a fixture has function scope so I'm suspect that if in future you used the same fixture for other tests then may you will get an error.

I don't know how `wrapanapi` handle this but I think one check is a better option (already disconnected or not).

Reply:
Thanks Nikhil, on second thought this doesn't really need to be a fixture, it is very specific to the `test_no_dvd_ruins_refresh` test, so I'll put it in there.